docs/userguide/datafiles.rst

```diff
@@ -2,13 +2,14 @@
 Data Files Support
 ====================
 
-Old packaging installation methods in the Python ecosystem
-have traditionally allowed installation of "data files", which
-are placed in a platform-specific location.  However, the most common use case
-for data files distributed with a package is for use *by* the package, usually
-by including the data files **inside the package directory**.
-
-Setuptools focuses on this most common type of data files and offers three ways
+Old packaging installation methods in the Python ecosystem have
+traditionally allowed the inclusion of "data files" (files beyond
+:ref:`the default set <manifest>` ), which are placed in a platform-specific
+location. However, the most common use case for data files distributed
+with a package is for use *by* the package, usually by including the
+data files **inside the package directory**.
+
+``Setuptools`` focuses on this most common type of data files and offers three ways
```

Comment:
> Thank you for your suggestion. I understand that your main objective was to clarify what “data files” are.

Yes exactly. Because this is the starting sentence of the `Data Files Support` section, and we have to make sure users understand what is considered "data file" in the first place, such that they could decide whether they even need these keywords at all.

For example C source code/READMEs are not "Python" files, but we don't need to declare them to include them into the package. And therefore we might need to clarify the following definition of "data files":
> we define "data files" as non-Python files


> the proposed parenthesis seem to suggest: data file = file not included in the "default set". This could lead to confusion, as what defines a data file is not fundamentally related to whether a file is included in the default set or not. 

Thanks a ton for the clarification! Yes my sketch indeed would need more discussion and check.

> They may also be referred to as "resource files."

Personally I think the following terminologies are already very confusing: "package data", "data file", "source distribution", "files inside the wheel", and their difference is pretty unclear, and can we somehow define/clarify them, and perhaps avoid introducing more synonyms ("resource files" in this case)?

Reply:
I have no problems with omitting `resource files`, I just mentioned before because other packages started calling them that way (e.g. `importlib_resources`). Please feel free to use and modify my suggestion to align with your vision in the contribution.

----

> For example C source code/READMEs are not "Python" files, but we don't need to declare them to include them into the package.

Yes, in this case C/README files are not data files, because they don't fit in the remaining part of the suggested definition:

```rst
... that are installed alongside Python modules and packages on the user's machine
when they install a :term:`distribution <Distribution Package>` from PyPI
or via a ``.whl`` file.
These files are typically intended for use at runtime by the package itself or
to influence the behavior of other packages or systems.
```

C-files are not installed along side Python files in the user's machine and also are not intended to be used in runtime, right?

----

> I just noticed [this](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/) from official Python docs:
> 
> > MANIFEST.in does not affect binary distributions such as wheels.


That is not 100% precise is it? `MANIFEST.in` determines what goes into the `sdist`, and then the contents of the `sdist` influence what goes into the wheel (specially when `include-package-data = true` which is the default)... So there is a potential indirect effect there. That is because the build process works more or less like in the mermaidjs diagram below[^1]:

```mermaid
graph LR
    src[fa:fa-laptop-code source code] --> b1(fa:fa-hammer)
    
    subgraph build process
      b1 --> sdist(fa:fa-file-code sdist) --> b2(fa:fa-hammer)
      b2 --> wheel(fa:fa-box wheel)
       
      build([build dependencies]) --> b1
      build([build dependencies]) --> b2
    end

    subgraph installation process
        wheel --> pkg
        pkg(fa:fa-box-open) --> inst[fa:fa-python installed packages]
        runtime([runtime dependencies]) --> inst
    end

    classDef deps fill:#2aa198,stroke:#333
    classDef env fill:#6c71c4,stroke:#333
    build:::deps
    runtime:::deps
```

[^1]: The "build process" creates distribution artifacts/packages. `sdist` distributions are meant to be platform independent (but may contain varying levels of optimisation - e.g. compiling Python to C via Cython). `wheel` distributions contain the final files meant to be directly copied to the user's machine, and therefore may be platform specific.